### PR TITLE
update autovalue

### LINKED ISSUE #66: Gradle 5.0 deprecation warning: annotation processor path
> Task :compileJava UP-TO-DATE
Detecting annotation processors on the compile classpath has been deprecated. Gradle 5.0 will ignore annotation processors on the compile classpath. The following annotation processors were detected on the compile classpath: 'com.google.auto.value.processor.AutoAnnotationProcessor' and 'com.google.auto.value.processor.AutoValueBuilderProcessor' and 'com.google.auto.value.processor.AutoValueProcessor'.  Please add them to the annotation processor path instead. If you did not intend to use annotation processors, you can use the '-proc:none' compiler argument to ignore them.

### LINKED ISSUE #100: Upgrade autovalue to 1.7
The automatic update failed. Someone will need to look into this:

> Task :compileJava
warning: [options] bootstrap class path not set in conjunction with -source 1.7
/home/travis/build/googleapis/api-common-java/src/main/java/com/google/api/pathtemplate/PathTemplate.java:34: error: package com.google.auto.value does not exist
import com.google.auto.value.AutoValue;
                            ^
/home/travis/build/googleapis/api-common-java/src/main/java/com/google/api/pathtemplate/PathTemplate.java:171: error: cannot find symbol
  @AutoValue
   ^
  symbol:   class AutoValue
  location: class PathTemplate
2 errors

> Task :compileJava FAILED

### PR DESCRIPTION
**Pull request by elharo**
@chingor13 

fixes #66 
fixes #100

### CODE REVIEW

**Comment by elharo** [comment]
Fails on JDK 7:

BUILD FAILED in 7s
1 actionable task: 1 executed

[31;1mThe command "eval ./gradlew assemble " failed. Retrying, 3 of 3.[0m

> Task :compileJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> com/google/auto/value/extension/memoized/processor/MemoizedValidator : Unsupported major.minor version 52.0

File: build.gradle [hunk 1]
```diff
@@ -57,12 +58,15 @@ repositories {
...
+  annotationProcessor libraries.auto_value
+
   compile libraries.guava,
     libraries.jsr305,
     libraries.javax_annotations
 
-  compileOnly libraries.autovalue,
-    libraries.error_prone_annotations
+  compileOnly libraries.error_prone_annotations,
+    libraries.auto_value_annotations
```
Comment (chingor13) [inline build.gradle]: I think auto_value is the compile only dependency and auto_value_annotations is an actual dependency

File: build.gradle [hunk 2]
```diff
@@ -134,9 +138,9 @@ configurations {
 }
 
 dependencies {
-  codeGeneration libraries.autovalue, libraries.jsr305
+  codeGeneration libraries.auto_value, libraries.jsr305
   compile libraries.jsr305
-  compileOnly libraries.autovalue
+  compileOnly libraries.auto_value_annotations
```
Comment (chingor13) [inline build.gradle]: Same here

File: build.gradle (hunk 1 above)
Comment (elharo) [reply build.gradle]: done for annotation. auto_value should be on the annotationProcessor path.

File: build.gradle (hunk 2 above)
Comment (elharo) [reply build.gradle]: done

**Pull request merged**